src/hypothesis/strategies.py

```diff
@@ -982,32 +987,61 @@ def fractions(min_value=None, max_value=None, max_denominator=None):
 
     If max_denominator is not None then the absolute value of the denominator
     of any generated values is no greater than max_denominator. Note that
-    max_denominator must be at least 1.
+    max_denominator must be None or a positive integer, and the denominator
```

Comment:
> deciding whether there exists a rational number of a designated denominator between two rational bounds

Note that the problem isn't a single designated denominator - that's easy. You just multiply both bounds by the denominator, round the bottom up to an integer and the top down to an integer, and if bounds are still in the right order then you've got a rational of that denominator between them (anything in that integer range divided by the denominator).

The problem scenario is finding one of *bounded* denominator without checking every value `<= n`.